### PR TITLE
5.x merge 4.x

### PR DESCRIPTION
**Pull request by asmorkalov**
OpenCV Extra: https://github.com/opencv/opencv_extra/pull/1210
Contrib: https://github.com/opencv/opencv_contrib/pull/3795

#25785 from refmitchell:issue_25784
#25901 from mshabunin:fix-riscv-aarch-baseline
#26088 from plctlab:rvp_pt2
#26109 from WanliZhong:univ_intrin_operator2warpper
#26113 from FantasqueX:zlib-ng-2-2-1
#26132 from asmorkalov:as/tlile_leakyRelu
#26134 from savuor:rv/mixed_arithm_channels
#26143 from asmorkalov:as/HAL_opticalFlowLK
#26146 from mshabunin:fix-test-overrides
#26148 from mshabunin:fix-sift-corruption
#26152 from sturkmen72:m_buf_supported
#26153 from onurcankaraman:sample/trackerUpdate
#26154 from cabelo:yolov5l

Previous "Merge 4.x": #26135

### CODE REVIEW

**Comment by asmorkalov** [comment]
@mshabunin please take a look. I dropped some 5.x specific flags for RISC-V in favor of new approach with toolchain files. Please ensure that it works as you expect now.
Also there are documentation warnings in 5.x branch. I fixed them in https://github.com/opencv/opencv/pull/26177.

File: cmake/OpenCVCompilerOptimizations.cmake [hunk 1]
```diff
@@ -390,26 +415,13 @@ elseif(PPC64LE)
   set(CPU_BASELINE "VSX" CACHE STRING "${HELP_CPU_BASELINE}")
 
 elseif(RISCV)
-  if(NOT DEFINED PLATFORM_STR)
-    set(PLATFORM_STR "rv64gc")
-  endif()
 
-  ocv_update(CPU_KNOWN_OPTIMIZATIONS "RVV;FP16;RVV_ZVFH")
```
Comment (mshabunin) [inline cmake/OpenCVCompilerOptimizations.cmake:397]: We need to keep FP16 and RVV_ZVFH in the list of known optimizations (line 55, line 397) as well as TEST_FILE, IMPLIES, FLAGS_ON variables (lines 399-407).

I'll adapt flags-riscv64.cmake accordingly after the merge.

Comment (asmorkalov) [reply cmake/OpenCVCompilerOptimizations.cmake:397]: @mshabunin Fixed.

Comment (asmorkalov) [reply cmake/OpenCVCompilerOptimizations.cmake:397]: Done.

**Review by mshabunin — APPROVED** [review]

**Pull request merged**